api_app/analyzers_manager/file_analyzers/cape_sandbox.py

```diff
@@ -0,0 +1,232 @@
+        task_id,
+    ) -> dict:
+        status_api = self.__base_url + "/apiv2/tasks/status/" + str(task_id)
+        for i in range(self.max_tries):
+            logger.info(
+                f"[POLLING] (Job: {self.job_id} -> "
+                f"CAPEsandbox __poll_for_result #{i + 1}/{self.max_tries}"
+            )
+            try:
+                r = self.__session.get(status_api)
+                if r.status_code == 429:
+                    logger.info(
+                        f"[INFO] (Job: {self.job_id} -> "
+                        "Rate limited by CAPESandbox API. "
+                        f"Sleeping for {self.poll_distance} seconds."
+                    )
+                    time.sleep(self.poll_distance)
+                else:
+                    r.raise_for_status()
+            except requests.RequestException as e:
+                if i == self.max_tries - 1:
+                    raise AnalyzerRunException(e)
+                    # ^ Unhandled exception at the last trial raised.
+                logger.info(
+                    f"[WARNING] (Job: {self.job_id} -> "
+                    f"Unhandled exception at poll attempt number: "
+                    f"{i}/{self.max_tries}. Ignoring and proceeding.."
+                )
+
+            responded_json = r.json()
+            error = responded_json.get("error")
+            data = responded_json.get("data")
+
+            logger.info(
+                f"[INFO] (Job: {self.job_id} -> "
+                f"Status of the CAPESandbox task: {data}"
+            )
+
+            if error:
+                raise AnalyzerRunException(error)
+
+            elif data in ("reported", "completed"):
+                report_url = (
+                    self.__base_url
+                    + "/apiv2/tasks/get/report/"
+                    + str(task_id)
+                    + "/json"
+                )
+                try:
+                    final_request = self.__session.get(
+                        report_url,
+                    )
+                    final_request.raise_for_status()
+                except requests.RequestException as e:
+                    raise AnalyzerRunException(e)
+
+                final_json = final_request.json()
+                response_ = final_json
+                return response_
+
+            elif data in ("pending", "running", "processing"):
+                if i == self.max_tries - 1:
+                    raise AnalyzerRunException(f"Task still {data}. Polling has ended.")
+                    """
+                        At times, What I have noticed is that
+                        CAPESandbox likes to leave your submission pending.
+                        It has a proper pending queue.
+
+                        Usually the status changes like this:
+                        pending -> running -> processing -> completed -> reported
+                    """
+                logger.info(
+                    f"[INFO] (Job: {self.job_id} ->"
```

Comment:
if you put the log in this case, you should put it also in the previous one. 

You need to imagine that, with the logs, it is possible to completely understand the flow of the code. If there are possible paths without logs, it is not acceptable and it would create confusion to the person who is debugging the code

Reply:
Understood. I left it there assuming that celery logs would anyway report the log as completed right after this because the last condition is the condition where the outputs are supposed to be returned. But I will add it there anyway.